test/test_pebble.py

```diff
@@ -2736,6 +3000,94 @@ def setUp(self):
 
         self.client = pebble.Client(socket_path=socket_path)
 
+    def test_checks_and_health(self):
+        checks = self.client.get_checks()
+        self.assertTrue(len(checks) == 0 or len(checks) == 3, checks)
+
+        self.client.add_layer('layer', {
+            'checks': {
+                'bad': {
+                    'override': 'replace',
+                    'level': 'ready',
+                    'period': '50ms',
+                    'threshold': 2,
+                    'exec': {
+                        'command': 'sleep x',
```

Comment:
Maybe this is a dumb question - but `sleep x` should exit nonzero right - so why does your assert below say the status is `UP`?

Reply:
It's only marked as `DOWN` after two checks have taken place (and the number of failures has hit the threshold). Which happens after 2 times the 50ms period. So initially the check will be `UP`, and only down after two tries, which is tested in the loop below. I've added comments here to clarify -- thanks.